### ISSUE TITLE
Rule Change: `capitalized-comments` should support way more "ignore" comments

### ISSUE THREAD

**Issue by fregante** [eslint/eslint#20927]
### What rule do you want to change?

capitalized-comments

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
// svelte-ignore custom_element_props_identifier
// biome-ignore lint/suspicious/noDuplicateProperties
// prettier-ignore
// codespell:ignore
```

### What does the rule currently do for this code?

They're all reported: https://eslint.org/play/#eyJ0ZXh0IjoiLy8gc3ZlbHRlLWlnbm9yZSBjdXN0b21fZWxlbWVudF9wcm9wc19pZGVudGlmaWVyXG4vLyBiaW9tZS1pZ25vcmUgbGludC9zdXNwaWNpb3VzL25vRHVwbGljYXRlUHJvcGVydGllc1xuLy8gcHJldHRpZXItaWdub3JlXG4vLyBjb2Rlc3BlbGw6aWdub3JlIiwib3B0aW9ucyI6eyJydWxlcyI6eyJjYXBpdGFsaXplZC1jb21tZW50cyI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### What will the rule do after it's changed?

None of them are reported

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

This is the relevant regex:

https://github.com/eslint/eslint/blob/b0e466b6ab47bfc7de43d8de0c315d8ee83aa584/lib/rules/utils/ast-utils.js#L45-L46

It could be updated to match `/^\w+[-:]?ignore:? /` to avoid having to maintain an ever-growing list of tools.

Webpack also has a number of magic comments, which I'm surprised nobody already added them here:

https://webpack.js.org/api/module-methods/#magic-comments

This is [xo's config for the rule](https://github.com/xojs/eslint-config-xo/blob/8951bad342d1d60889c017ff1444e37a9e521b6b/source/javascript-rules.js#L287) for example:

```js
		/pragma|ignore|prettier-ignore|biome-ignore|codespell:ignore|webpack\w+:|c8|v8|type-coverage:/v,
```

**Comment by Tanujkanti4441** [comment]
Hi @fregante, thanks for the suggestion, but as you can see, there is already an option `ignorePattern` to avoid reporting the comments you have mentioned in the issue when needed, and also, this rule is [frozen](https://eslint.org/docs/latest/contribute/core-rules#frozen-rules), so we cannot apply this suggestion to `capitalized-comments`.

**Comment by fregante** [comment]
I literally linked to a configuration that has the option, the request is specifically to have good defaults. 

It's crazy to me that you accept such a bad default that requires most people to customize it. Why not remove rules you don't want to support? Let @eslint-stylistic take it.
